pkg/sql/pgwire/conn.go

```diff
@@ -610,10 +610,18 @@ func (c *conn) sendInitialConnData(
 	// defaults with client-provided values.
 	// For details see: https://www.postgresql.org/docs/10/static/libpq-status.html
 	for _, param := range statusReportParams {
+		param := param
 		value := connHandler.GetStatusParam(ctx, param)
 		if err := c.sendStatusParam(param, value); err != nil {
 			return sql.ConnectionHandler{}, err
 		}
+		// pgwire also expects updates when these parameters change.
+		connHandler.RegisterOnSessionDataChange(param, func(val interface{}) {
+			convertVal := val.(string)
```

Comment:
Can val be a `string` instead of `interface{}` to avoid the forced cast? Seems unsafe otherwise.